[mtt] ERROR: failed to create rc QP TX: Invalid argument

Configuration
```
OMPI: 4.0.0rc5
MOFED: MLNX_OFED_LINUX-4.4-2.0.7.0
Module: hpcx-gcc (2018-11-15)
Test module: mtt-tests/hpcx-gcc
Nodes: hercules x98 (ppn=32(x98), nodelist=clx-hercules-[031-128])
```

MTT log: http://e2e-gw.mellanox.com:4080/hpc/scrap/users/mtt/scratch/ucx_ompi/20181117_153304_6132_232975_clx-hercules-031/html/test_stdout_5ZroVS.txt

Cmd:
`mpirun -np 3136 --display-map -mca btl self --tag-output --timestamp-output -mca pml ucx -mca coll '^hcoll' --bind-to core -x UCX_NET_DEVICES=mlx5_2:1,mlx5_3:1 -mca osc '^ucx' -x UCX_IB_REG_METHODS=rcache,direct -x UCX_TLS=all -x UCX_RC_TM_ENABLE=n -mca pmix_base_async_modex 0 -mca mpi_add_procs_cutoff 100000 -mca pmix_base_collect_data 1 -x UCX_MAX_EAGER_LANES=3 -x UCX_MAX_RNDV_LANES=3 -x UCX_IB_TRAFFIC_CLASS=106 -x UCX_IB_GID_INDEX=auto --map-by node /hpc/local/benchmarks/hpcx_install_2018-11-15/mtt-tests-gcc/installs/uuGb/tests/mpich_tests/mpich-mellanox.git/test/mpi/comm/dupic`

Output:
```sh
Sat Nov 17 16:06:38 2018[1,2842]<stderr>:[clx-hercules-031:22693] pml_ucx.c:323 Error: ucp_ep_create(proc=1521) failed: Input/output error
Sat Nov 17 16:06:38 2018[1,2842]<stderr>:[clx-hercules-031:22693] [[5804,1],2842] ORTE_ERROR_LOG: Error in file communicator/comm.c at line 1661
Sat Nov 17 16:06:38 2018[1,2842]<stderr>:[clx-hercules-031:22693] 1274: Error in ompi_get_rprocsSat Nov 17 16:06:38 2018[1,2842]<stderr>:[clx-hercules-031:22693:0:22693] Caught signal 11 (Segmentation fault: address not mapped to object at address 0xf8)
Sat Nov 17 16:06:38 2018[1,2548]<stderr>:[clx-hercules-031:22686] pml_ucx.c:323 Error: ucp_ep_create(proc=1227) failed: Input/output error
Sat Nov 17 16:06:38 2018[1,2548]<stderr>:[clx-hercules-031:22686] [[5804,1],2548] ORTE_ERROR_LOG: Error in file communicator/comm.c at line 1661
Sat Nov 17 16:06:38 2018[1,2548]<stderr>:[clx-hercules-031:22686] 980: Error in ompi_get_rprocs
Sat Nov 17 16:06:38 2018[1,2548]<stderr>:[clx-hercules-031:22686:0:22686] Caught signal 11 (Segmentation fault: address not mapped to object at address 0xf8)
Sat Nov 17 16:06:38 2018[1,2842]<stdout>:[1542463598.313258] [clx-hercules-031:22693:0]       ib_iface.c:571  UCX  ERROR iface=0x790c00: failed to create rc QP TX wr:256 sge:3 inl:64 RX wr:0 sge:1 inl 64: Invalid argument
Sat Nov 17 16:06:38 2018[1,2548]<stdout>:[1542463598.313444] [clx-hercules-031:22686:0]       ib_iface.c:571  UCX  ERROR iface=0x790c40: failed to create rc QP TX wr:256 sge:3 inl:64 RX wr:0 sge:1 inl 64: Invalid argument

...

Wed Nov 21 08:08:57 2018[1,79]<stderr>:==== backtrace ====
Wed Nov 21 08:08:57 2018[1,79]<stderr>: 0 0x00000000000628a0 PMPI_Comm_set_name()  ???:0
Wed Nov 21 08:08:57 2018[1,79]<stderr>: 1 0x00000000000628a0 PMPI_Comm_size()  /hpc/local/benchmarks/hpcx_install_2018-11-19/src/hpcx-gcc-redhat7.4/ompi/ompi/mpi/c/profile/pcomm_size.c:63
Wed Nov 21 08:08:57 2018[1,79]<stderr>: 2 0x00000000004040a3 MTestGetIntercomm()  ???:0
Wed Nov 21 08:08:57 2018[1,79]<stderr>: 3 0x00000000004028d7 main()  ???:0
Wed Nov 21 08:08:57 2018[1,79]<stderr>: 4 0x00000000000223d5 __libc_start_main()  ???:0
Wed Nov 21 08:08:57 2018[1,79]<stderr>: 5 0x0000000000402519 _start()  ???:0
Wed Nov 21 08:08:57 2018[1,79]<stderr>:===================
Wed Nov 21 08:08:57 2018[1,1106]<stdout>:[1542780536.019597] [clx-hercules-058:23991:0]       ib_iface.c:571  UCX  ERROR iface=0x78f860: failed to create rc QP TX wr:256 sge:3 inl:64 RX wr:0 sge:1 inl 64: Invalid argument
Wed Nov 21 08:08:57 2018[1,1498]<stdout>:[1542780536.022668] [clx-hercules-058:23995:0]       ib_iface.c:571  UCX  ERROR iface=0x78f8a0: failed to create rc QP TX wr:256 sge:3 inl:64 RX wr:0 sge:1 inl 64: Invalid argument
Wed Nov 21 08:08:57 2018[1,1239]<stdout>:[1542780535.901276] [clx-hercules-093:27615:0]       ib_iface.c:571  UCX  ERROR iface=0x78f880: failed to create rc QP TX wr:256 sge:3 inl:64 RX wr:0 sge:1 inl 64: Invalid argument
Wed Nov 21 08:08:57 2018[1,428]<stdout>:[1542780535.964728] [clx-hercules-066:23670:0]       ib_iface.c:571  UCX  ERROR iface=0x790000: failed to create rc QP TX wr:256 sge:3 inl:64 RX wr:0 sge:1 inl 64: Invalid argument
```

Test works well if only one device is used:
```sh
$mpirun -np 3136  -mca btl self --tag-output --timestamp-output -mca pml ucx -mca coll '^hcoll' --bind-to core -x UCX_NET_DEVICES=mlx5_2:1 -mca osc '^ucx' -x UCX_IB_REG_METHODS=rcache,direct -x UCX_TLS=all -x UCX_RC_TM_ENABLE=n -mca pmix_base_async_modex 0 -mca mpi_add_procs_cutoff 100000 -mca pmix_base_collect_data 1 -x UCX_MAX_EAGER_LANES=3 -x UCX_MAX_RNDV_LANES=3 -x UCX_IB_TRAFFIC_CLASS=106 -x UCX_IB_GID_INDEX=auto --map-by node /hpc/local/benchmarks/hpcx_install_2018-11-15/mtt-tests-gcc/installs/uuGb/tests/mpich_tests/mpich-mellanox.git/test/mpi/comm/dupic
Wed Nov 21 08:09:29 2018[1,0]<stdout>: No Errors

$mpirun -np 3136  -mca btl self --tag-output --timestamp-output -mca pml ucx -mca coll '^hcoll' --bind-to core -x UCX_NET_DEVICES=mlx5_3:1 -mca osc '^ucx' -x UCX_IB_REG_METHODS=rcache,direct -x UCX_TLS=all -x UCX_RC_TM_ENABLE=n -mca pmix_base_async_modex 0 -mca mpi_add_procs_cutoff 100000 -mca pmix_base_collect_data 1 -x UCX_MAX_EAGER_LANES=3 -x UCX_MAX_RNDV_LANES=3 -x UCX_IB_TRAFFIC_CLASS=106 -x UCX_IB_GID_INDEX=auto --map-by node /hpc/local/benchmarks/hpcx_install_2018-11-15/mtt-tests-gcc/installs/uuGb/tests/mpich_tests/mpich-mellanox.git/test/mpi/comm/dupic
Wed Nov 21 08:09:49 2018[1,0]<stdout>: No Errors
```

It can be related to non-lossless configuration of ROCE on Hercules.

Test passed with NP=2650. Failed with NP=2660.

--- a/src/ucp/wireup/select.c
+++ b/src/ucp/wireup/select.c
@@ -955,7 +955,9 @@ static double ucp_wireup_am_bw_score_func(ucp_context_h context,
     double size = iface_attr->cap.am.max_bcopy;
     double time = (size / ucs_min(iface_attr->bandwidth,
                                   remote_iface_attr->bandwidth)) +
-                  iface_attr->overhead + remote_iface_attr->overhead;
+                  iface_attr->overhead + remote_iface_attr->overhead +
+                  ucp_wireup_tl_iface_latency(context, iface_attr, remote_iface_attr);
+
     return size / time * 1e-5;
 }
 

